### ISSUE TITLE
Handle semicolon properly

### ISSUE THREAD

**Issue by yhara** [shiika-lang/shiika#1]
Currently `Lexer` does not distinguish `;` from newline, but there is a case where newline is allowed but semicolon is not. 

Example:

```
# OK
foo(1,
2)
# NG
foo(1,;2)
```

**Comment by yamakoud** [comment]
Should we implement Token::Newline and Token::Semicolon first?

Do you have any other suggestions?

**Comment by yamakoud** [comment]
@yhara 
Should we implement Token::Newline and Token::Semicolon first?

Do you have any other suggestions?

**Comment by yhara** [comment]
I don't have a particular idea to fix this but yes, I think we need to distinguish newline and semicolon at a token level.

**Comment by yamakoud** [comment]
I got it. Thank you!

**Comment by yamakoud** [comment]
I'll add some cases.

```
error
Test.new.foo(;1,2)
Test.new.foo(1;,2)
Test.new.foo(1,;2)
Test.new.foo(1,2;)

success
Test.new.foo(1,2);
```

**Comment by yamakoud** [comment]
plz assign me if you can

**Comment by yamakoud** [comment]
The PR is ready for review

Plz review this when it's convenient

**Comment by yhara** [comment]
fixed by #534.
